File: app/src/main/java/com/example/linkshare/MainActivity.java
package com.example.linkshare;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {


    private Button btnCrear;
    private Button btnLista;

    private DatabaseReference tDatabase;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnCrear = findViewById(R.id.btnCrear);
        btnLista = findViewById(R.id.btnLista);


        tDatabase = FirebaseDatabase.getInstance().getReference();

        btnCrear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(MainActivity.this, NewLinkActivity.class);
                startActivity(myIntent);
            }
        });

        btnLista.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(MainActivity.this, ListActivity.class);
                startActivity(myIntent);
            }
        });
    }
}

File: app/src/main/java/com/example/linkshare/MainActivity.kt
package com.example.linkshare

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.database.FirebaseDatabase

class MainActivity : AppCompatActivity() {

    lateinit var editTextName: EditText
    lateinit var btnCrearDatos: Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editTextName = findViewById(R.id.editTitulo)
        btnCrearDatos = findViewById(R.id.btnCrearDatos)

        btnCrearDatos.setOnClickListener{
            saveFirebaseData()
        }
    }

    private fun saveFirebaseData(){
        var tituloText = editTextName.text.toString().trim()

        if(tituloText.isEmpty()){
            tituloText = "No hay titulo"
        }


        val ref = FirebaseDatabase.getInstance().getReference("titulos")
        val tituloId = ref.push().key
        val titulo = Titulo(tituloId, tituloText)

        if (tituloId != null) {
            ref.child(tituloId).setValue(titulo).addOnCompleteListener{
                Toast.makeText(applicationContext, "Titulo saved", Toast.LENGTH_SHORT)
            }
        }

        return
    }
}